src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingXmlDocumentSetInsecureResolverTests.cs

```diff
@@ -0,0 +1,455 @@
+			Try
+				doc.XmlResolver = Nothing
+			Catch
+				Throw
+			End Try
+		End Sub
+	End Class
+End Namespace");
+        }
+
+        [Fact]
+        public void XmlDocumentNoCtorUseNonSecureResolverInCatchClauseShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod(XmlDocument doc)
+        { 
+            try {   }
+            catch { 
+                doc.XmlResolver = new XmlUrlResolver();
+            }
+            finally {}
+        }
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(12, 17, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Private Shared Sub TestMethod(doc As XmlDocument)
+            Try
+            Catch
+                doc.XmlResolver = New XmlUrlResolver()
+            Finally
+            End Try
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(9, 17, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentNoCtorUseNonSecureResolverInFinallyClauseShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod(XmlDocument doc)
+        { 
+            try {   }
+            catch { throw; }
+            finally {
+                doc.XmlResolver = new XmlUrlResolver();
+            }
+        }
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(13, 17, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Private Shared Sub TestMethod(doc As XmlDocument)
+            Try
+            Catch
+                Throw
+            Finally
+                doc.XmlResolver = New XmlUrlResolver()
+            End Try
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(11, 17, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentNoCtorDontSetResolverShouldNotGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod(XmlDocument doc, XmlReader reader)
+        {
+            doc.Load(reader);
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+	Class TestClass
+		Private Shared Sub TestMethod(doc As XmlDocument, reader As XmlReader)
+			doc.Load(reader)
+		End Sub
+	End Class
+End Namespace");
+        }
+
+        [Fact]
+        public void XmlDocumentNoCtorUseNonSecureResolverShouldGenerateDiagnostic()
+        {
```

Comment:
This seems to be the one test you actually need. Why do you need the tests above, where you show that it's also bad to do this within a `catch` block or a `finally` block?